packages/plugin-core/src/_extension.ts

```diff
@@ -138,43 +136,30 @@ async function postReloadWorkspace() {
   Logger.info({ ctx, msg: "exit" });
 }
 
-async function startServer() {
-  const ctx = "startServer";
+async function startServerProcess(): Promise<{
+  port: number;
+  subprocess?: ExecaChildProcess;
+}> {
   const { nextServerUrl, nextStaticRoot, engineServerPort } =
     getWS().config.dev || {};
+  // const ctx = "startServer";
   const maybePort =
     DendronWorkspace.configuration().get<number | undefined>(
       CONFIG.SERVER_PORT.key
     ) || engineServerPort;
-  const logPath = DendronWorkspace.instance().context.logPath;
-  Logger.info({ ctx, logLevel: process.env["LOG_LEVEL"], maybePort });
-  if (!maybePort) {
-    const { launchv2 } = require("@dendronhq/api-server"); // eslint-disable-line global-require
-    return launchv2({
-      port: maybePort,
-      logPath: path.join(logPath, "dendron.server.log"),
-      nextServerUrl,
-      nextStaticRoot,
-    });
+  const port = maybePort;
+  if (port) {
+    return { port };
   }
-  return { port: maybePort };
-}
-
-// @ts-ignore
-function subscribeToPortChange() {
-  const ctx = "subscribeToPortChange";
-  HistoryService.instance().subscribe(
-    "apiServer",
-    async (event: HistoryEvent) => {
-      if (event.action === "changedPort") {
-        const port = DendronWorkspace.serverConfiguration().serverPort;
-        const engine = WSUtils.updateEngineAPI(port);
-        await engine.init();
-        Logger.info({ ctx, msg: "fin init Engine" });
-        await reloadWorkspace();
-      }
-    }
-  );
+  const logPath = DendronWorkspace.instance().context.logPath;
+  const out = await ServerUtils.execServerNode({
```

Comment:
I'm guessing that setting debugger statements in the engine will no longer break if this is a separate process? If so, maybe we need to build out a developer mode vs production mode.

Some random ideas from previous projects: in dev mode, we can detect if an existing server is running. If not, then instantiate one. If it does exist, then connect to it.

We can instantiate the debug server via another instance of VS Code. In that case, breakpoints should hit in the server too?

Reply:
good point. we already do that for the integration tests because it takes too long to spawn a separate process otherwise. will update

Reply:
updated